src/CSET/operators/regrid.py

```diff
@@ -0,0 +1,168 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Operators to regrid cubes."""
+
+
+import iris
+import iris.cube
+import numpy as np
+
+
+def regrid_onto_cube(
+    incube: iris.cube.Cube, target: iris.cube.Cube, method: str, **kwargs
+) -> iris.cube.Cube:
+    """Regrid a cube, projecting onto a target cube.
+
+    Cube must have at least 2 dimensions.
+
+    Arguments
+    ----------
+    incube: Cube
+        An iris cube of the data to regrid. As a minimum, it needs to be 2D with a latitude,
+        longitude coordinates.
+    target: Cube
+        An iris cube of the data to regrid onto. It needs to be 2D with a latitude,
+        longitude coordinate.
+    method: str
+        Method used to regrid onto, etc. Linear will use iris.analysis.Linear()
+
+    Returns
+    -------
+    iris.cube.Cube
+        An iris cube of the data that has been regridded.
+
+    Raises
+    ------
+    ValueError
+        If a unique x/y coordinate cannot be found
+    NotImplementedError
+        If the cubes grid, or the method for regridding, is not yet supported.
+    """
+    # Usual names for spatial coordinates
+    X_COORD_NAMES = ["longitude", "grid_longitude", "projection_x_coordinate", "x"]
+    Y_COORD_NAMES = ["latitude", "grid_latitude", "projection_y_coordinate", "y"]
+
+    # Get a list of coordinate names for the cube
+    coord_names = [coord.name() for coord in incube.coords()]
+
+    # Check which x-coordinate we have, if any
+    x_coords = [coord for coord in coord_names if coord in X_COORD_NAMES]
+    if len(x_coords) != 1:
+        raise ValueError("Could not identify a unique x-coordinate in cube")
+    x_coord = incube.coord(x_coords[0])
+
+    # Check which y-coordinate we have, if any
+    y_coords = [coord for coord in coord_names if coord in Y_COORD_NAMES]
+    if len(y_coords) != 1:
+        raise ValueError("Could not identify a unique y-coordinate in cube")
+    y_coord = incube.coord(y_coords[0])
+
+    # List of supported grids - check if it is compatible
+    supported_grids = (iris.coord_systems.GeogCS,)
+    if isinstance(incube.coord(x_coord).coord_system, supported_grids):
+        raise NotImplementedError(
+            f"Does not currently support {incube.coord(x_coord).coord_system} regrid method"
+        )
+    elif isinstance(incube.coord(y_coord).coord_system, supported_grids):
+        raise NotImplementedError(
+            f"Does not currently support {incube.coord(y_coord).coord_system} regrid method"
+        )
```

Comment:
```suggestion
    if not isinstance(incube.coord(x_coord).coord_system, supported_grids):
        raise NotImplementedError(
            f"Does not currently support {incube.coord(x_coord).coord_system} regrid method"
        )
    if not isinstance(incube.coord(y_coord).coord_system, supported_grids):
        raise NotImplementedError(
            f"Does not currently support {incube.coord(y_coord).coord_system} regrid method"
        )
```
We don't want to say supported grids are not supported.
The elif is not needed as the raise will have already ended the function if the first conditional is True.